src/combinatorics/simple_graph/inc_matrix.lean

```diff
@@ -0,0 +1,388 @@
+respect to a given `orientation`.
+
+## Main results
+
+1. `adj_sum_of_mul_inc_one`: `∑ e : E, M i e * M j e = 1`, for any two adjacent vertices
+`i` and `j`.
+2. `inc_matrix_mul_non_adj`: `M i e * M j e = 0`, for any two distinct non-adjacent vertices
+`i`, `j` and edge `e`.
+3. `inc_matrix_element_power_id`: Every element from `M` is idempotent.
+4. `degree_equals_sum_of_incidence_row`: For any vertex `i`, the sum on the ith row of `M` is
+equal to the degree of `i`.
+5. `oriented_inc_matrix_elem_squared`: `(N(o) i e) ^ 2 = M i e`, for any orientation `o`,
+vertex `i` and edge `e`.
+6. `oriented_inc_matrix_mul_of_adj`: For any adjacent vertices `i`, `j` and edge `e`,
+`N(o) i e * N(o) j e = if e = ⟦(i, j)⟧ then -1 else 0`.
+7. `oriented_inc_matrix_mul_non_adj`: For any non-adjacent distinct vertices `i`, `j` and
+edge `e`, `N(o) i e * N(o) j e = 0`.
+8. `vec_mul_oriented_inc_matrix`: `(xᵀ ⬝ N(o)) e = x o.head(e) - x o.tail(e).`
+
+## References
+
+<https://en.wikipedia.org/wiki/Orientation_(graph_theory)>
+-/
+
+open_locale big_operators matrix
+open finset matrix simple_graph sym2
+
+universes u v
+variables {R : Type u} {V : Type v}
+
+section to_move
+
+@[simp]
+lemma ite_mul_ite_zero_right
+  {R : Type*} [mul_zero_class R] {P Q : Prop} [decidable P] [decidable Q] (a b : R):
+  (ite P a 0) * (ite Q b 0) = ite (P ∧ Q) (a * b) 0 :=
+by simp [←ite_and]
+
+end to_move
+
+namespace simple_graph
+
+variables (G : simple_graph V)
+
+-- ### Helping lemmas for edges
+
+lemma edge_set_ne {u v : V} {e : G.edge_set} (h : ↑e = ⟦(u, v)⟧) : u ≠ v :=
+begin
+  apply G.ne_of_adj,
+  rw [←G.mem_edge_set, ←h],
+  exact e.property
+end
+
+lemma incidence_equiv {i : V} {e : G.edge_set} : ↑e ∈ G.incidence_set i ↔ i ∈ (↑e : sym2 V) :=
+by simp only [incidence_set, true_and, set.mem_sep_eq, subtype.coe_prop]
+
+lemma any_vertex_if_incidence_set {i u v : V} :
+  ⟦(u, v)⟧ ∈ G.incidence_set i → i = u ∨ i = v :=
+by simp only [←mem_iff, incidence_set, set.mem_sep_eq, and_imp, imp_self, implies_true_iff]
+
+lemma incidence_set_if_any_vertex {i u v : V} (h : ⟦(u, v)⟧ ∈ G.edge_set) :
+  i = u ∨ i = v → ⟦(u, v)⟧ ∈ G.incidence_set i :=
+by simp only [←mem_iff, incidence_set, set.mem_sep_eq, h, true_and, imp_self]
+
+lemma edge_in_two_incidence_sets {i j : V} {e : sym2 V} (H_ne : i ≠ j) :
+  e ∈ G.incidence_set i ∧ e ∈ G.incidence_set j → e = ⟦(i, j)⟧ :=
+begin
+  refine quotient.rec_on_subsingleton e (λ p, _),
+  rcases p with ⟨v, w⟩,
+  rintros ⟨⟨-, H_i⟩, ⟨-, H_j⟩⟩,
+  rw ←elems_iff_eq H_ne,
+  tauto,
+end
+
+lemma mem_incidence_sets_iff_eq {i j : V} {e : sym2 V} (h : G.adj i j) :
+  e ∈ G.incidence_set i ∧ e ∈ G.incidence_set j ↔ e = ⟦(i, j)⟧ :=
+begin
+ split,
+  { apply edge_in_two_incidence_sets _,
+  rintro rfl,
+  exact G.loopless i h },
+  { rintro rfl,
+  nth_rewrite 1 eq_swap,
+  simp only [h, G.sym h, mem_incidence_set, and_self] }
+end
+
+lemma adj_iff_exists_edge_coe {i j : V} : G.adj i j ↔ ∃ (e : G.edge_set), ↑e = ⟦(i, j)⟧ :=
```

Comment:
I'm not sure you need this.  You can get the element of `G.edge_set` directly like so:
```
example {i j : V} (h : G.adj i j) : G.edge_set := ⟨⟦(i, j)⟧, h⟩
```

Reply:
Not sure how I can replace `adj_iff_exists_edge_coe` in `adj_sum_of_mul_inc_one` though

Reply:
It's actually pretty easy. cf my other comment.